### ISSUE TITLE
Lora train loss stay above 1

### ISSUE THREAD

**Issue by gladjoyhub** [ml-explore/mlx-examples#216]
I was using a custom dataset that I could achieve loss of 0.000 on OpenAI fine tuning base model gpt-3.5-turbo-1106, which they charge around 2 dollors. So I am switching to MLX lora training base model Mistral-7B-V0.1 on my M1 max 32GB for free, but I can't get loss under 1, and the answer accuracy is bad. How to improve loss? Also, the generated adapter_model.npz is just 1.7MB for 4 lora-layers, 3.4MB for 8 lora-layers. I got adapter size of 700MB on Google Colab. Am I missing something? Thanks!

Iter 370: Train loss 1.146, It/sec 1.238, Tokens/sec 87.913
Iter 380: Train loss 1.187, It/sec 1.266, Tokens/sec 90.614
Iter 390: Train loss 1.153, It/sec 1.291, Tokens/sec 90.374
Iter 400: Train loss 1.229, It/sec 1.235, Tokens/sec 87.933
Iter 400: Val loss 1.495, Val took 12.619s
Iter 410: Train loss 1.167, It/sec 1.297, Tokens/sec 92.735
Iter 420: Train loss 1.270, It/sec 1.238, Tokens/sec 90.773
Iter 430: Train loss 1.112, It/sec 1.221, Tokens/sec 87.397
Iter 440: Train loss 1.111, It/sec 1.219, Tokens/sec 86.669
Iter 450: Train loss 1.149, It/sec 1.296, Tokens/sec 92.377
Iter 460: Train loss 1.150, It/sec 1.244, Tokens/sec 87.858
Iter 470: Train loss 1.114, It/sec 1.246, Tokens/sec 85.948
Iter 480: Train loss 1.155, It/sec 1.238, Tokens/sec 87.634
Iter 490: Train loss 1.200, It/sec 1.279, Tokens/sec 89.657
Iter 500: Train loss 1.145, It/sec 1.187, Tokens/sec 84.166
Iter 510: Train loss 1.346, It/sec 1.183, Tokens/sec 93.086
Iter 520: Train loss 1.231, It/sec 1.226, Tokens/sec 89.856
Iter 530: Train loss 1.191, It/sec 1.216, Tokens/sec 87.157
Iter 540: Train loss 1.162, It/sec 1.144, Tokens/sec 80.527
Iter 550: Train loss 1.097, It/sec 1.249, Tokens/sec 86.462
Iter 560: Train loss 1.283, It/sec 1.211, Tokens/sec 88.662
Iter 570: Train loss 1.109, It/sec 0.546, Tokens/sec 38.088
Iter 580: Train loss 1.340, It/sec 1.256, Tokens/sec 94.858
Iter 590: Train loss 1.211, It/sec 1.274, Tokens/sec 93.270
Iter 600: Train loss 1.137, It/sec 1.236, Tokens/sec 85.811
Iter 600: Val loss 1.475, Val took 12.621s

**Comment by awni** [comment]
>  I got adapter size of 700MB on Google Colab. Am I missing something?

I'm not sure what other parameters they include in the adapter

> but I can't get loss under 1, and the answer accuracy is bad. How to improve loss?

Does using more LoRA layers help? If so maybe use more? If not 🤔 ... maybe there is a problem with the data format or learning parameters.  It does look like your train loss is well below your validation loss...when you say the "answer accuracy is bad" how did you test it?

**Comment by awni** [comment]
Closing for now as inactive. Let me know if you still need help and I can reopen.
